### PR TITLE
1.4.x refactor

### PR DESCRIPTION
**Pull request by yama**

### CODE REVIEW

**Review by Pathologic — COMMENTED** [review]
https://github.com/evocms-community/evolution/pull/222/commits/4a44275ad711840cb4f67914b2d1b6bfc4641f45

1.4.x branch is for bug fixes, new features should be synced with 3.x

what's the use case of this feature?

**Comment by Pathologic** [comment]
@yama please, separate large PR to small ones, because it's hard to review.

**Comment by yama** [comment]
> please, separate large PR to small ones, because it's hard to review.

OK, I'll do that next time.
thanks

**Comment by yama** [comment]
Our business needs to use 1.4.x for another 2 years in business. In parallel, we will also maintain 3.x.

**Comment by Pathologic** [comment]
> Our business needs to use 1.4.x for another 2 years in business. In parallel, we will also maintain 3.x.

1.4.x will be supported as long as possible, but we've decided to focus efforts on future versions. So new features are implemented in 3.x first.

**Comment by yama** [comment]
@Pathologic

ok, I fully understand that. However, I still need 1.4.x.
I understand what you are saying. Please delete this pull request if it is not needed by this project. thank you.

**Comment by Pathologic** [comment]
I'll check it anyway.

File: manager/includes/document.parser.class.inc.php [hunk 1]
```diff
@@ -456,9 +461,21 @@ public function getSettings()
         $this->config['valid_hostnames'] = MODX_SITE_HOSTNAMES;
         $this->config['site_manager_url'] = MODX_MANAGER_URL;
         $this->config['site_manager_path'] = MODX_MANAGER_PATH;
-        $this->error_reporting = isset($this->config['error_reporting']) ? $this->config['error_reporting'] : 0;
-        $this->config['filemanager_path'] = str_replace('[(base_path)]', MODX_BASE_PATH, (isset($this->config['filemanager_path']) ? $this->config['filemanager_path'] : ''));
-        $this->config['rb_base_dir'] = str_replace('[(base_path)]', MODX_BASE_PATH, (isset($this->config['rb_base_dir']) ? $this->config['rb_base_dir'] : ''));
+        $this->error_reporting = $this->config['error_reporting'] ?? 0;
+        if(strpos($this->config['filemanager_path'], '[(base_path)]')!==false) {
```
Comment (64j) [inline manager/includes/document.parser.class.inc.php:465]: Undefined array key "filemanager_path"

File: manager/includes/document.parser.class.inc.php [hunk 2]
```diff
@@ -456,9 +461,21 @@ public function getSettings()
...
+            $this->config['filemanager_path'] = str_replace(
+                '[(base_path)]',
+                MODX_BASE_PATH,
+                $this->config['filemanager_path']
+            );
+        }
+        if(strpos($this->config['rb_base_dir'],'[(base_path)]')!==false) {
```
Comment (64j) [inline manager/includes/document.parser.class.inc.php:472]: Undefined array key "rb_base_dir"

File: manager/includes/document.parser.class.inc.php [hunk 3]
```diff
@@ -614,8 +629,16 @@ public function getUserSettings()
         }
 
         $this->config = array_merge($this->config, $usrSettings);
-        $this->config['filemanager_path'] = str_replace('[(base_path)]', MODX_BASE_PATH, $this->config['filemanager_path']);
-        $this->config['rb_base_dir'] = str_replace('[(base_path)]', MODX_BASE_PATH, $this->config['rb_base_dir']);
+        $this->config['filemanager_path'] = str_replace(
+            '[(base_path)]',
+            MODX_BASE_PATH,
+            $this->config['filemanager_path']
```
Comment (64j) [inline manager/includes/document.parser.class.inc.php:635]: Undefined array key "filemanager_path"

File: manager/includes/document.parser.class.inc.php [hunk 4]
```diff
@@ -614,8 +629,16 @@ public function getUserSettings()
...
+        );
+        $this->config['rb_base_dir'] = str_replace(
+            '[(base_path)]',
+            MODX_BASE_PATH,
+            $this->config['rb_base_dir']
```
Comment (64j) [inline manager/includes/document.parser.class.inc.php:640]: Undefined array key "rb_base_dir"

**Comment by Pathologic** [comment]
Well. This PR is too large and hard to test; it seems there are no fixes of any bugs, mostly  code style improvements. So I decided to reject it, though I respect the work done. I'll try to create a contribution guide to avoid such unpleasant situations.

**Comment by yama** [comment]
@Pathologic 
ok, I try again today. thanks.

**Pull request closed without merging**